crowbar_framework/app/models/service_object.rb

```diff
@@ -1492,14 +1492,15 @@ def chef_daemon(action, node_list)
   private
 
   def wait_for_chef_clients(node_name, options = {})
-    options = if options.fetch(:logger)
+    use_logger = options.fetch(:logger)
+    options = if use_logger
       {:logger => @logger}
     else
       {}
     end
-    @logger.debug("wait_for_chef_clients: Waiting for already running chef-clients on #{node_name}.")
+    @logger.debug("wait_for_chef_clients: Waiting for already running chef-clients on #{node_name}.") if use_logger
```

Comment:
What is the reason behind options mangling?

Wouldn't something like this work, considering `wait_for_chef_clients` is called with `options = { :logger => @logger }`?:

```ruby
logger = options.fetch(:logger) { Rails.logger }
logger.debug("wait_for...")
...
options[:logger] ||= logger
unless RemoteNode.chef_ready?(....options)
  logger.error("Waiting...")
```